### DISCUSSION TITLE
Request Response Time [V5]

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#2890] [Ideas]
Hi @thetutlage ,
We need to calculate request response time. It's helpful for APIs log.
Currently we using ```execution-time``` npm package or we are missing some feature?

Thanks.

**Comment by Egorka** [comment]
Hi @pk4all,

I use a slightly modified [LogRequest](https://docs.adonisjs.com/guides/middleware#middleware-classes) middleware:

```
import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'

export default class LogRequest {
  
  public async handle(
    { request, response }: HttpContextContract,
    next: () => Promise<void>
  ) {

    const start_req = new Date().getTime()

    response.response.on('finish', () => {

      const end_req = new Date().getTime() - start_req

      console.log(`-> ${request.method()}: ${request.url()} | ${response.response.statusCode} | ${end_req} ms`)

    })

    await next()

  }
}
```

**Reply by pk4all** [reply]
Hi,
How to get response data after request complete?

**Reply by feniljariwala82** [reply]
@Egorka showed the best way to solve this problem. @pk4all you can use this middleware on any route. And it will calculate response time internally. You will get data once you define it.
